solver/src/driver.rs

```diff
@@ -32,16 +34,25 @@ impl Driver {
                 None => return Ok(()),
                 Some(settlement) => settlement,
             };
-        // TODO: encode settlement, check if we need to approve spending to uniswap
+        // TODO: check if we need to approve spending to uniswap
         // TODO: use retry transaction sending crate for updating gas prices
-        self.contract
-            .settle_call(&settlement)
-            .await
-            .context("settle call failed")?;
-        self.contract
-            .settle_send(&settlement)
-            .await
-            .context("settle send failed")?;
+        let settle = || {
+            self.settlement.settle(
+                settlement.tokens(),
+                settlement.clearing_prices(),
+                settlement.fee_factor,
+                settlement
+                    .encode_trades()
+                    .expect("naive solver created invalid settlement"),
+                settlement.encode_interactions(
+                    &self.uniswap_router.address(),
+                    &self.settlement.address(),
+                ),
+                Vec::new(),
+            )
+        };
+        settle().call().await.context("settle call failed")?;
+        settle().send().await.context("settle send failed")?;
```

Comment:
nit can we in the context call it simulating settle vs executing sett (the difference between send and call might not be imediately obvious when reading this in a larger log context)